SQF/dayz_code/compile/keyboard.sqf

```diff
@@ -2,331 +2,337 @@
 
 #include "\ca\editor\Data\Scripts\dikCodes.h"
 
-_dikCode = _this select 1;
+_dikCode    = _this select 1;
 _shiftState = _this select 2;
-_ctrlState = _this select 3;
-_altState = _this select 4;
-_handled = false;
+_ctrlState  = _this select 3;
+_altState   = _this select 4;
+_handled    = false;
 
 if (isNil "keyboard_keys") then {
-    _cancelBuild = {
-		DZE_cancelBuilding = true;
-		call dayz_EjectPlayer;
-		_handled = false;
-		if (r_player_dead) then {_handled = true;}; // Disable ESC after death
+  _cancelBuild = {
+    DZE_cancelBuilding = true;
+    call dayz_ejectPlayer;
+    if (r_player_dead) then {_handled = true;} else {_handled = false;};
+  };
+  _dze_f = {
+    if (!_ctrlState && !_altState) then {DZE_F = true;};
+  };
+  _dze_q = {
+    if (!_ctrlState && !_altState) then {DZE_Q = true;};
+    if (!_ctrlState && _altState) then {DZE_Q_alt = true;};
+    if (_ctrlState && !_altState) then {DZE_Q_ctrl = true;};
+  };
+  _dze_z = {
+    if (!_ctrlState && !_altState) then {DZE_Z = true;};
+    if (!_ctrlState && _altState) then {DZE_Z_alt = true;};
+    if (_ctrlState && !_altState) then {DZE_Z_ctrl = true;};
+  };
+  _autoRun = {
+    if (!dayz_autoRun) then {
+      dayz_autoRun = true;
+      dayz_autoRunThread = [] spawn {
+        _weapon = currentWeapon player;
+        while {dayz_autoRun} do {
+          //SurfaceIsWater doesn't work for ponds
+          //Check Weapon to detect ArmA action (dayz action is handled in dz_fn_switchWeapon)
```

Comment:
This will probably take a lot of performance due to several statements on an if in a loop. May increase the uiSleep?